cmd/cluster/utils.go

```diff
@@ -224,3 +227,163 @@ func getBaseVersionFromUnstable(version string) (string, error) {
+	nodes, allocatableCPU, allocatableMemory, err := getClusterAllocatableResources()
+	if err != nil {
+		return "", deployment.AstarteClusterProfile{}, err
+	}
+
+	fmt.Printf("Cluster has %v nodes\n", nodes)
+	fmt.Printf("Allocatable CPU is %vm\n", allocatableCPU)
+	fmt.Printf("Allocatable Memory is %v\n", bytefmt.ByteSize(uint64(allocatableMemory)))
+	fmt.Println()
+
+	clusterRequirements := deployment.AstarteProfileRequirements{
+		CPUAllocation:    allocatableCPU,
+		MemoryAllocation: allocatableMemory,
+		MinNodes:         nodes,
+		MaxNodes:         nodes,
+	}
+	availableProfiles := deployment.GetProfilesForVersionAndRequirements(astarteVersion, clusterRequirements)
+
+	if len(availableProfiles) == 0 {
+		return "", deployment.AstarteClusterProfile{}, fmt.Errorf("Unfortunately, your cluster allocatable resources do not allow for any profile to be deployed")
+	}
+
+	unexcludedAvailableProfiles := map[string]deployment.AstarteClusterProfile{}
+	// Handle exclusion list
+	for k, v := range availableProfiles {
+		exclude := false
+		for _, s := range excluding {
+			if s == k {
+				exclude = true
+				break
+			}
+		}
+		if exclude {
+			continue
+		}
+
+		unexcludedAvailableProfiles[k] = v
+	}
+
+	if len(unexcludedAvailableProfiles) == 0 {
+		return "", deployment.AstarteClusterProfile{}, fmt.Errorf("There are no other profiles which can be deployed on this cluster besides %s", excluding)
+	}
+
+	fmt.Println("You can safely deploy the following Profiles on this cluster:")
+	for _, v := range unexcludedAvailableProfiles {
+		fmt.Printf("%s: %s\n", v.Name, v.Description)
+	}
+
+	fmt.Println()
+	profile := getStringFlagFromPromptOrDie(command, "profile", "Which profile would you like to deploy?", "", false)
+
+	if _, ok := unexcludedAvailableProfiles[profile]; !ok {
+		fmt.Printf("Profile %s does not exist! Aborting.\n", profile)
+		os.Exit(1)
+	}
+
+	return profile, availableProfiles[profile], nil
+}
+
+func promptForProfile(command *cobra.Command, astarteVersion *semver.Version) (string, deployment.AstarteClusterProfile, error) {
+	return promptForProfileExcluding(command, astarteVersion, []string{})
+}
+
+func getValueFromSpec(spec map[string]interface{}, field string) interface{} {
+	fieldTokens := strings.Split(field, ".")
+	aMap := spec
+
+	for i, f := range fieldTokens {
+		if _, ok := aMap[f]; ok {
+			switch v := aMap[f].(type) {
+			case map[string]interface{}:
+				aMap = v
+			case string:
+				if i == len(fieldTokens)-1 {
+					return v
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func getStringFromSpecOrFlag(spec map[string]interface{}, field string, command *cobra.Command, flagName string) string {
+	ret := getValueFromSpec(spec, field)
+	if ret != nil {
+		return ret.(string)
+	}
+
+	fl, err := command.Flags().GetString(flagName)
+	if err != nil {
+		// If the flag isn't defined, assume it's not a big deal.
+		if !strings.HasPrefix(err.Error(), "flag accessed but not defined") {
```

Comment:
This is brittle since it relies on the text of the error.
Check if the flag exists before doing GetString with https://godoc.org/github.com/spf13/pflag#FlagSet.Lookup instead